### ISSUE TITLE
Support for Godot-gdscript-toolkit

### ISSUE THREAD

**Issue by Arrow-x** [jose-elias-alvarez/null-ls.nvim#826]
### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues) and there are no existing ones with the same request.

### Feature description

supersedes #368
This project https://github.com/Scony/godot-gdscript-toolkit have Linting and Formating utility that I want to use with null-ls but when I add them to my config it says 
```
failed to load builtin gdformat for method formatting; please check your config 
failed to load builtin gdlint for method diagnostics; please check your config
```

### Help

Yes, but I don't know how to start. I would need guidance

**Comment by jose-elias-alvarez** [comment]
Yep, you can see from [our master list](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/c832a0ecb18fac8b35967111327434edf6f782aa/doc/BUILTINS.md) that we don't have any support for these sources currently. Happy to review a PR adding support (formatters in particular are generally pretty easy) but I'll also mention that nvim-lspconfig [has support for a language server](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#gdscript), which should provide a better experience.

**Comment by Arrow-x** [comment]
this is an initial implementation #827
